### DISCUSSION TITLE
آموزش استفاده از ریلیتی در مرزبان

### DISCUSSION THREAD

**Discussion by mahanmi** [Gozargah/Marzban#231] [Show and tell]
سلام
اول از همه باید وصل سرور بشید و وارد فایل xray_config.json خودتون بشید
و در قسمت inbounds 
این رو اضافه کنید
(البته الان با قابلیت جدید مرزبان میشه این فایل رو داخل داشبورد هم ادیت کرد)
```
{
            "tag": "VLESS_TCP_REALITY_ZULA",
            "listen": "0.0.0.0",
            "port": 2083,
            "protocol": "vless",
            "settings": {
                "clients": [],
                "decryption": "none"
            },
            "streamSettings": {
                "network": "tcp",
                "security": "reality",
                "realitySettings": {
                    "show": false,
                    "dest": "zula.ir:443",
                    "xver": 0,
                    "serverNames": [
                        "zula.ir",
                        "www.zula.ir",
                        "account.zula.ir",
                        "cdn-download2.zula.ir",
                        "cdn-download1.zula.ir",
                        "cdn-download.zula.ir"
                    ],
                    "privateKey": "qEUE0DeGWCMvddo1iRNvNcoF1xBVg1XKebazOzeVVVw",
                    "publicKey": "O1hStZ1z9SZyUYipBJ_BE-IV1o4ucrSULW65sMEaB1g",
                    "shortIds": [
                        "ea83a8b0"
                    ]
                }
            },
            "sniffing": {
                "enabled": true,
                "destOverride": [
                    "http",
                    "tls"
                ]
            }
        }
```
بعدش با کامند های زیر وارد
docker exec 
بشید و
publickey و privatekey
خودتون رو جنریت کنید و بجای چیزی که من توی متن گذاشتم بزارید
```
docker exec marzban-marzban-1 xray x25519
```
یه همچین چیزی بهتون میده
![image](https://user-images.githubusercontent.com/101924479/235518729-adbfe81c-26bc-4f1e-9e62-de9ad4f5a831.png)
و در آخر باید پورتی که انتخاب کردید ریلیتی رو روش ران کنید به داکر وصل کنید که کار بکنه
برای این کار باید وارد 
docker-compose.yml
تون بشید و
توی بخش
ports:
پورتتونو بزنید برای من 2083 بود پس میزنم
![image](https://user-images.githubusercontent.com/101924479/235537454-d87c8572-0e2d-4d15-84cf-bdc161c27682.png)
حالا هم
docker compose restart 
و خلاص
من بازم یه فایل کامل نمونه که شامل: 
VMESS-WS
VLESS-TCP-REALITY
رو که خودم استفاده میکنم این پایین براتون آپلود میکنم
[xray_config.zip](https://github.com/Gozargah/Marzban/files/11367736/xray_config.zip)

**Comment by lucifersanthem** [comment]
داخل docker-compose.yml پورت نیست برای من

**Reply by SaintShit** [reply]
> داخل docker-compose.yml پورت نیست برای من

اگر `network_mode: host` هست توی فایلتون نیاز نیست اون بخش پورت رو انجام بدین

**Comment by mrfishing1** [comment]
چگونه باید مقدار flow هم اضافه کنیم میشه توضیح بدید ؟

**Reply by mahanmi** [reply]
داخل پنل وقتی vless انتخاب میکنید گزینش هست

**Reply by silvermanq** [reply]
یعنی هیچ راهی نیست از طریق هسته انجام بشه ؟

**Comment by morteza-rashidi** [comment]
اون مورد مربوط به جنریت public key private key رو میزنم با این خطا روبرو میشم
service "marzban-marzban-1" is not running container

**Reply by mahanmi** [reply]
بجای marzban-marzban-1 اسم کانتینری که در حال ران هست رو بزنید اسمشو میتونید با docker ps ببینید
